From: Fabian Maurer <dark.shadow4@web.de>
Subject: [PATCH v2 3/3] msi: Show messageboxes as modal
Message-Id: <20210911201224.78002-3-dark.shadow4@web.de>
Date: Sat, 11 Sep 2021 22:12:24 +0200
In-Reply-To: <20210911201224.78002-1-dark.shadow4@web.de>
References: <20210911201224.78002-1-dark.shadow4@web.de>

This last patch is a bit different.
It fixes an issue, but I don't know enough about msi internals to tell how correct it is.
If you don't mind, please take a look :)

Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=51718
---
 dlls/msi/dialog.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/dlls/msi/dialog.c b/dlls/msi/dialog.c
index 193ab84afd4..8f5e12e0d84 100644
--- a/dlls/msi/dialog.c
+++ b/dlls/msi/dialog.c
@@ -3885,6 +3885,10 @@ static LRESULT WINAPI MSIDialog_WndProc( HWND hwnd, UINT msg,
         {
             EnableWindow(dialog->parent->hwnd, TRUE);
         }
+        if (gUIhwnd == hwnd)
+        {
+            gUIhwnd = NULL;
+        }
         dialog->hwnd = NULL;
         return 0;
     case WM_NOTIFY:
@@ -3932,6 +3936,11 @@ static UINT dialog_run_message_loop( msi_dialog *dialog )
         return ERROR_FUNCTION_FAILED;
     }

+    if (!dialog->parent && !gUIhwnd)
+    {
+        gUIhwnd = hwnd;
+    }
+
     if (parent)
     {
         EnableWindow(parent, FALSE);
--
2.33.0



